### PR TITLE
Clean up dialog and consolidate node styling

### PR DESCRIPTION
**Pull request by blink1073**
Refactors the dialog to be a lot more consistent and adds tests.
Pulls out node styling functions and uses them for button, select, and input nodes in the app.

### CODE REVIEW

File: src/common/dialog.ts [hunk 1]
```diff
@@ -1,319 +1,127 @@
...
-  options = options || {};
-  let host = options.host || document.body;
-  options.host = host;
-  options.body = options.body || '';
-  // NOTE: This code assumes only one dialog is shown at the time:
-  okButton.text = options.okText ? options.okText : 'OK';
-  options.buttons = options.buttons || [cancelButton, okButton];
-  if (!options.buttons.length) {
-    options.buttons = [okButton];
+  show(): Promise<Dialog.IButton> {
```
Comment (afshin) [inline src/common/dialog.ts]: Does there need to be a check here?
```
if (this._promise) {
  return this._promise.promise;
}
```

File: src/common/dialog.ts [hunk 2]
```diff
@@ -443,171 +251,426 @@ class Dialog extends Panel {
...
 
+    /**
+     * Create the button nodes of the dialog.
+     *
+     * @param buttons - The buttons data.
+     *
+     * @returns An array of elements for the buttons.
+     */
+    createButtonNodes(buttons: ReadonlyArray<IButton>): ReadonlyArray<HTMLElement> {
+      let nodes = new Array<HTMLElement>(buttons.length);
```
Comment (afshin) [inline src/common/dialog.ts]:
```
return buttons.map(node => VirtualDOM.realize(this.renderButtonNode(node)));
```
😈

File: src/common/dialog.ts [hunk 3]
```diff
@@ -443,171 +251,426 @@ class Dialog extends Panel {
...
-  child.classList.add(BODY_CONTENT_CLASS);
-  return new Widget({node: child});
-}
 
+    /**
+     * Render a button node for the dialog.
+     *
+     * @param data - The data to use for rendering the button.
+     *
+     * @returns A virtual element representing the button.
```
Comment (afshin) [inline src/common/dialog.ts]: I'm not sure I understand the benefit of using virtual DOM here if it gets immediately `realize`d.

File: src/common/dialog.ts (hunk 1 above)
Comment (blink1073) [inline src/common/dialog.ts]: yep

**Pull request merged**